### ISSUE TITLE
Validate region trees to be strict DAGs on construction

### ISSUE THREAD

**Issue by dickscheid** [FZJ-INM1-BDA/siibra-python#201]
Some region trees seem to have inconsistent definitions and lead to regions containing themselves as a child region. This must not happen and needs to be fixed, and validated on object construction.

@xgui3783 this is a reminder to ourselves from the discussion this morning.

**Comment by AhmetNSimsek** [comment]
> also, unfortunately, I believe @AhmetNSimsek spoke with sebastian and co, and duplicated parent and child is a product of the pipeline, and cannot be easily fixed automatically (please correct me if I am wrong @AhmetNSimsek )

You are correct. This happens when a leaf region does not have enough parents and the mappers have to enter another parent in the table they track this. The solution they employed was making a parent be a child of itself in terminology json while in the actual region tree, these two regions re just one region. Sebastian and Hartmut are aware of this issue but not sure this will change.
